testnet/stacks-node/src/mockamoto.rs

```diff
@@ -0,0 +1,565 @@
+        let new_snapshot = make_snapshot(&parent_snapshot, &miner_pk_hash)?;
+        let mut sortdb_tx = self.sortdb.tx_handle_begin(&parent_snapshot.sortition_id)?;
+
+        sortdb_tx.append_chain_tip_snapshot(
+            &parent_snapshot,
+            &new_snapshot,
+            &vec![],
+            &vec![],
+            None,
+            None,
+            None,
+        )?;
+
+        sortdb_tx.commit()?;
+
+        let staging_db_tx = self.chainstate.db_tx_begin()?;
+        NakamotoChainState::set_burn_block_processed(
+            &staging_db_tx,
+            &new_snapshot.burn_header_hash,
+        )?;
+        staging_db_tx.commit()?;
+
+        Ok(())
+    }
+
+    fn mine_stacks_block(&mut self) -> Result<NakamotoBlock, ChainstateError> {
+        let sortition_tip = SortitionDB::get_canonical_burn_chain_tip(self.sortdb.conn())?;
+        let chain_id = self.chainstate.chain_id;
+        let (mut chainstate_tx, clarity_instance) = self.chainstate.chainstate_tx_begin().unwrap();
+
+        let (is_genesis, chain_tip_bh, chain_tip_ch) = match NakamotoChainState::get_chain_tip(
+            &chainstate_tx,
+            &sortition_tip.consensus_hash,
+            self.sortdb.conn(),
+        ) {
+            Ok(chain_tip) => (false, chain_tip.0, chain_tip.1),
+            Err(ChainstateError::NoSuchBlockError) =>
+            // No stacks tip yet, parent should be genesis
+            {
+                (
+                    true,
+                    FIRST_STACKS_BLOCK_HASH,
+                    FIRST_BURNCHAIN_CONSENSUS_HASH,
+                )
+            }
+            Err(e) => return Err(e),
+        };
+
+        let (parent_chain_length, parent_burn_height) = if is_genesis {
+            (0, 0)
+        } else {
+            let tip_block_id = StacksBlockId::new(&chain_tip_ch, &chain_tip_bh);
+            let tip_info = NakamotoChainState::get_block_header(&chainstate_tx, &tip_block_id)?
+                .ok_or(ChainstateError::NoSuchBlockError)?;
+            (tip_info.stacks_block_height, tip_info.burn_header_height)
+        };
+
+        let miner_nonce = 2 * parent_chain_length;
+
+        let coinbase_tx_payload = TransactionPayload::Coinbase(CoinbasePayload([1; 32]), None);
+        let mut coinbase_tx = StacksTransaction::new(
+            TransactionVersion::Testnet,
+            TransactionAuth::from_p2pkh(&self.miner_key).unwrap(),
+            coinbase_tx_payload,
+        );
+        coinbase_tx.chain_id = chain_id;
+        coinbase_tx.set_origin_nonce(miner_nonce);
+        let mut coinbase_tx_signer = StacksTransactionSigner::new(&coinbase_tx);
+        coinbase_tx_signer.sign_origin(&self.miner_key).unwrap();
+        let coinbase_tx = coinbase_tx_signer.get_tx().unwrap();
+
+        let parent_block_id = StacksBlockId::new(&chain_tip_ch, &chain_tip_bh);
+        let tenure_change_tx_payload = TransactionPayload::TenureChange(TenureChangePayload {
```

Comment:
Can you add a comment here? I guess you're just including a tenure change in every block to test some more paths?